src/webview/chat.js

```diff
@@ -47,30 +47,45 @@
 
     if (message.command === "add_result") {
       addAssistantMessageToUI(message.result);
+    } else if (message.command === "recipe_request") {
+      sendRecipeRequest(message.result);
     } else if (message.command === "clear_chat") {
       clearAllMessages();
     } else if (message.command === "focus") {
       messageInput.focus();
     }
   });
-  function sendMessageToExtension(message) {
+  function sendRequestToExtension(message) {
+    addAssistantThinkingMessageToUI();
     vscode.postMessage({ type: "chat_request", data: message });
   }
 
   function clearAllMessages() {
-    currentAssistantMessage = null;
+    assistantMessageFinished();
     chatContainer.textContent = "";
   }
 
   function sendUserMessage() {
     const message = messageInput.value.trim();
     messageInput.value = "";
     addUserMessageToUI(message);
-    addAssistantThinkingMessageToUI();
-    sendMessageToExtension(message);
+    sendRequestToExtension(message);
     checkTextarea();
   }
 
+  function sendRecipeRequest(message) {
+    // Ensure we don't add on to the previous message
+    assistantMessageFinished();
+    addAssistantMessageToUI("Executing Recipe: " + message);
+    // Ensure new responses don't get added on to this one
+    assistantMessageFinished();
+    sendRequestToExtension(message);
```

Comment:
rather ugly right now - needs refactoring with a proper system for when messages are finished (this will also allow us to disable sending new messages while we're waiting)